### PR TITLE
add panoptic dataset

### PR DESCRIPTION
**Pull request by liuxin9608**
add panoptic dataset

### CODE REVIEW

File: tests/test_datasets/test_top_down_dataset.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def load_json_to_output(json_name):
             anno['keypoints'], dtype=np.float32).reshape((1, -1, 3))
         box = np.array([0, 0, 0, 0, 0, 0], dtype=np.float32).reshape(1, -1)
         img_path = []
-        img_path[:0] = image_info['file_name']
+        img_path[:0] = prefix + image_info['file_name']
```
Comment (innerlee) [inline tests/test_datasets/test_top_down_dataset.py:22]: use `os.path.join` when joining path

**Comment by innerlee** [comment]
looks good. is the train accuracy okay?

File: mmpose/datasets/datasets/top_down/topdown_panoptic_dataset.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+            if metric not in allowed_metrics:
+                raise KeyError(f'metric {metric} is not supported')
+
+        res_file = os.path.join(res_folder, 'result_keypoints.json')
+
+        kpts = []
+
+        for preds, boxes, image_path in outputs:
+            str_image_path = ''.join(image_path)
+            print(str_image_path, flush=True)
```
Comment (innerlee) [inline mmpose/datasets/datasets/top_down/topdown_panoptic_dataset.py:194]: is this for debug?

File: mmpose/datasets/datasets/top_down/topdown_panoptic_dataset.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+        """This encodes bbox(x,y,w,w) into (center, scale)
+
+        Args:
+            x, y, w, h
+
+        Returns:
+            center (np.ndarray[float32](2,)): center of the bbox (x, y).
+            scale (np.ndarray[float32](2,)): scale of the bbox w & h.
+        """
+        aspect_ratio = self.ann_info['image_size'][0] / self.ann_info[
```
Comment (innerlee) [inline mmpose/datasets/datasets/top_down/topdown_panoptic_dataset.py:155]: here's a shorter version, don't know if u like the style

```python
aspect_ratio = np.true_divide(*self.ann_info['image_size'])
```

Comment (jin-s13) [reply mmpose/datasets/datasets/top_down/topdown_panoptic_dataset.py:155]: The readability gets worse...

**Comment by jin-s13** [comment]
> looks good. is the train accuracy okay?

Yes, the training accuracy looks good. I will upload a model & update the readme soon.

File: mmpose/datasets/datasets/top_down/topdown_panoptic_dataset.py (hunk 3 above)
Comment (innerlee) [reply mmpose/datasets/datasets/top_down/topdown_panoptic_dataset.py:155]: true

**Review by innerlee — APPROVED** [review]

**Comment by innerlee** [comment]
need a following pr to update the repo's readme and the changelog

**Pull request merged**